app/models/access_control_entry.rb

```diff
@@ -1,4 +1,11 @@
 class AccessControlEntry < ApplicationRecord
   acts_as_tenant(:tenant)
   belongs_to :aceable, :polymorphic => true
+  has_many :access_control_permissions, :dependent => :destroy
+  has_many :permissions, :through => :access_control_permissions
+
+  def add_new_permissions(permissions)
+    new_permissions = permissions - self.permissions.map(&:name)
+    new_permissions.each { |name| self.permissions << Permission.find_by!(:name => name) }
```

Comment:
@syncrou Can we add a test to have an invalid permission and have it kick out the exception for the !. Not sure how that propagates up.

Reply:
The openapi parser does the validation so we don't need it at this level.